### PR TITLE
Work on linting for files in .gitignore

### PR DESCRIPTION
**Pull request by M-Zuber**
This is an attempt to start building the module discussed in #41 .

Currently I need input on more tests to write, and then the discussion can move onto implementation.

### CODE REVIEW

**Comment by basicallydan** [comment]
Niiiiice! Great start :D I'll have a better look at this later!

File: lib/app.js [hunk 1]
```diff
@@ -94,6 +95,14 @@
 				passes = passes.concat(report.passes);
 				failures = failures.concat(report.failures);
 
+				if(options.ignore || lintOptions.ignore) {
+					var ignoreOptions = {ignore:options.ignore};
+					ignoreOptions.ignore = merge(ignoreOptions.ignore, lintOptions.ignore);
```
Comment (M-Zuber) [inline lib/app.js]: I don't like whats going on here.
The actual linter should be updated to expect a more normal looking onject

File: lib/lintIgnore.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+var merge = require('merge');
+
+module.exports = function(tree, options) {
+	var passes = [];
+	var failures = [];
+	var resolveRelevance = function (feature) {
+		return (!feature.relevant || feature.relevant === true || feature.relevant());
+	};
+	var files = {};
+	files = merge(files, options.ignore);
```
Comment (M-Zuber) [inline lib/lintIgnore.js]: There is no real reason for `options.ignore` not to just be `ignores`

Comment (basicallydan) [inline lib/lintIgnore.js]: You mean because it suggests an object rather than a boolean?

Comment (M-Zuber) [inline lib/lintIgnore.js]: `ignore` is an object like so:
```js
ignore: {
 'No debug folder': {
  path: /^debug/i,
  shouldExist: false,
  type: 'tree'
 }
}
```
I use the keys of the ignore object, so having to go through `options` seems like an unneeded step

**Comment by M-Zuber** [comment]
Stuff looks good.
:shipit:

**Pull request merged**